public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -198,17 +195,30 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
         .pipe(
           single(), // unsubscribes automatically after first result
           filter((response: any) => (response.cancelled ? false : true)),
-          map((response: any) => {
-            const delta = this.processResult(response, query, target, queries.length);
+          mergeMap((response: any) => {
+            const data = this.processResult(response, query, target, queries.length);
             const state: DataStreamState = {
               key: `prometheus-${target.refId}`,
-              state: query.instant ? LoadingState.Loading : LoadingState.Done,
+              state: LoadingState.Loading,
               request: options,
-              delta,
+              data,
               unsubscribe: () => undefined,
             };
 
-            return state;
+            if (query.instant) {
```

Comment:
Hack until we implement can use sub requests.

Reply:
does not need sub requests -- just a different key

Reply:
note the changes in #16676 are not required for sub requests, that really just uses them for the mixed query datasource and also adds work to better report it in the editor UI.  Multi-request streaming is already supported, but no built-in datasources use it (yet)... partly because keeping in sync with explore